### PR TITLE
MINOR: Extract SockerServer inner classes to server module

### PR DESCRIPTION
**Pull request by mimaison**
All remaining classes depend on KafkaConfig so can't be extracted at the moment.

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

File: server/src/main/java/org/apache/kafka/network/quota/ConnectionQuotaEntity.java [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+ */
+package org.apache.kafka.network.quota;
+
+import java.util.Map;
+
+/**
+ * Interface for connection quota configuration. Connection quotas can be configured at the
+ * broker, listener or IP level.
+ */
+public interface ConnectionQuotaEntity {
```
Comment (chia7712) [inline server/src/main/java/org/apache/kafka/network/quota/ConnectionQuotaEntity.java]: It seems
1. the sub-classes are small
2. we don't use casting in code base
3. they are a kind of pojo.
Maybe we can merge all sub impls into `ConnectionQuotaEntity`. For example:
```java
public class ConnectionQuotaEntity {

    public static final String CONNECTION_RATE_SENSOR_NAME = "Connection-Accept-Rate";
    public static final String CONNECTION_RATE_METRIC_NAME = "connection-accept-rate";
    public static final String LISTENER_THROTTLE_PREFIX = "";
    public static final String IP_METRIC_TAG = "ip";
    public static final String IP_THROTTLE_PREFIX = "ip-";
    private static final long INACTIVE_SENSOR_EXPIRATION_TIME_SECONDS = TimeUnit.HOURS.toSeconds(1);

    public static ConnectionQuotaEntity listenerQuotaEntity(String listenerName) {
        return new ConnectionQuotaEntity(CONNECTION_RATE_SENSOR_NAME + "-" + listenerName,
                CONNECTION_RATE_SENSOR_NAME, Long.MAX_VALUE, Collections.singletonMap("listener", listenerName));
    }

    public static ConnectionQuotaEntity brokerQuotaEntity() {
        return new ConnectionQuotaEntity(CONNECTION_RATE_SENSOR_NAME,
                "broker-" + ConnectionQuotaEntity.CONNECTION_RATE_METRIC_NAME,
                Long.MAX_VALUE, Collections.emptyMap());
    }

    private final String sensorName;
    private final String metricName;
    private final long sensorExpiration;
    private final Map<String, String> metricTags;

    private ConnectionQuotaEntity(String sensorName, String metricName, long sensorExpiration, Map<String, String> metricTags) {
        this.sensorName = sensorName;
        this.metricName = metricName;
        this.sensorExpiration = sensorExpiration;
        this.metricTags = metricTags;
    }

    /**
     * The name of the sensor for this quota entity
     */
    public String sensorName() {
        return sensorName;
    }

    /**
     * The name of the metric for this quota entity
     */
    public String metricName() {
        return metricName;
    }

    /**
     * The duration in second to keep the sensor even if no new values are recorded
     */
    public long sensorExpiration() {
        return sensorExpiration;
    }

    /**
     * Tags associated with this quota entity
     */
    public Map<String, String> metricTags() {
        return metricTags;
    }
}
```

**Review by chia7712 — COMMENTED** [review]
@mimaison nice refactor. one comment is left. PTAL

File: server/src/main/java/org/apache/kafka/network/quota/ConnectionQuotaEntity.java (hunk 1 above)
Comment (mimaison) [reply server/src/main/java/org/apache/kafka/network/quota/ConnectionQuotaEntity.java]: That's a good point. I followed your suggestion and merged all classes into `ConnectionQuotaEntity`. Thanks!

**Review by chia7712 — APPROVED** [review]
LGTM

**Pull request merged**